### PR TITLE
1. Change the inspector to a virtual list.

### PR DESCRIPTION
**Pull request by jm-observer**

### CODE REVIEW

File: src/inspector/data.rs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        }
+    }
+
+    fn get_children(
+        &self,
+        next: &mut usize,
+        min: usize,
+        max: usize,
+        level: usize,
+    ) -> Vec<(usize, usize, CapturedData)> {
```
Comment (jrmoulton) [inline src/inspector/data.rs:154]: It looks like both of these functions are relatively expensive to compute. Is there a reason that these aren't just computed and stored in the init function? Then the get children function could just return a slice without needing to build a new vec of children elements.

**Review by jrmoulton — COMMENTED** [review]
What is the motivation for making this change?

**Comment by jm-observer** [comment]
> What is the motivation for making this change?

After upgrading Floem to version 0.2.0, using the inspector causes the program to overflow.

> It looks like both of these functions are relatively expensive to compute. Is there a reason that these aren't just computed and stored in the init function? Then the get children function could just return a slice without needing to build a new vec of children elements.

As far as I know, a virtual list is relatively inexpensive and dynamically calculated.

Additionally, the example of the virtual list is overly simplistic.

**Comment by jrmoulton** [comment]
I should've tested it first. I thought this would be slower than the old way with rebuilding a list of items every time the list refreshes but it's definitely faster.

**Pull request merged**